### ISSUE TITLE
[Bug]: Different Token returned for PushManager/Delegate

### ISSUE THREAD

**Issue by killer-frog** [shinyorg/shiny#676]
### Component

Push - Native

### What platform(s) are effected?

- [ ] iOS 12
- [ ] iOS 13
- [X] iOS 14
- [ ] Android 8
- [ ] Android 9
- [ ] Android 10
- [ ] Android 11
- [ ] UWP

### Steps To Reproduce

1. Set up Code in PushDelegate log OnTokenChanged 

2. Call 
```  c#
var result = ShinyHost.Resolve<IPushManager>().RequestAccess();
AppCenter.Log(result.RegistrationToken);
```
3. In the pushDelegate Log the Token.

``` c#
        public async Task OnTokenChanged(string token)
        {
              AppCenter.Log(token);
        }
```

### Expected Behavior

Both token have the same value

### Actual Behavior

**IOS**

- The Registration token from IPushManager.Request acesss was : 

`433942434542384236383245373634364134334646374434443033454339353031413941343333333342384634304636454332413755424146` this token was wrong and device received no notifications.

- The token from Push Delegate's OnTokenReceived was : `C9BCEB8B682E7646A43FF7D4D03EC9501A9A43333B8F40F6EC2A7FBAF62DB3DD` which when used was the correct token and device received push notifications

The main issue this caused was that I have to handle the receipt of a new token differently for iOS and android.  For Android, so I have something like:

``` c#
            var result = await _pushManager.RequestAccess();

            if (result.Status != AccessState.Available || string.IsNullOrWhiteSpace(result.RegistrationToken))
            {
                return string.Empty;
            }

            if (_deviceInfo.Platform == DevicePlatform.iOS) return result.RegistrationToken;

            await RegisterForNotifications(result.RegistrationToken);  // Server Registration
```

but then I also have to handle new tokens in the Push Delegate for iOS

``` c#
        public async Task OnTokenChanged(string token)
        {
            _pushService ??= FreshIOC.Container.Resolve<IPushService>();
            await _pushService.RegisterForNotifications(token);
         }
```

**Android**
Android does not seem to call the PushDelegate's OnTokenReceived, however the token received from the `RequestAccess` call was correct and push notifications worked.

### Exception or Log output

Using Shiny v2.2.0.2829

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by aritchie** [comment]
Thank you for reporting this, it will be fixed as part of 2.3

**Comment by aritchie** [comment]
This is resolved as part of the latest preview.  OnTokenChanged (now called OnTokenRefreshed) is not called as part of RequestAccess
